libparsec/crates/client/src/workspace/store/resolve_path.rs

```diff
@@ -646,3 +658,106 @@ pub(crate) async fn resolve_path_for_reparenting(
+            super::GetManifestError::InvalidKeysBundle(err) => {
+                ResolvePathError::InvalidKeysBundle(err)
+            }
+            super::GetManifestError::InvalidCertificate(err) => {
+                ResolvePathError::InvalidCertificate(err)
+            }
+            super::GetManifestError::InvalidManifest(err) => ResolvePathError::InvalidManifest(err),
+            super::GetManifestError::Internal(err) => err.context("cannot retrieve path").into(),
+        })?
+        .parent();
+
+    // Loop until we reach the root
+    while current_entry_id != current_parent_id {
+        // Get the parent manifest
+        let parent_manifest =
+            store
+                .get_manifest(current_parent_id)
+                .await
+                .map_err(|err| match err {
+                    super::GetManifestError::Offline => ResolvePathError::Offline,
+                    super::GetManifestError::Stopped => ResolvePathError::Stopped,
+                    super::GetManifestError::EntryNotFound => ResolvePathError::EntryNotFound,
+                    super::GetManifestError::NoRealmAccess => ResolvePathError::NoRealmAccess,
+                    super::GetManifestError::InvalidKeysBundle(err) => {
+                        ResolvePathError::InvalidKeysBundle(err)
+                    }
+                    super::GetManifestError::InvalidCertificate(err) => {
+                        ResolvePathError::InvalidCertificate(err)
+                    }
+                    super::GetManifestError::InvalidManifest(err) => {
+                        ResolvePathError::InvalidManifest(err)
+                    }
+                    super::GetManifestError::Internal(err) => {
+                        err.context("cannot retrieve path").into()
+                    }
+                })?;
+
+        // A parent manifest should always be a folder
+        let parent_manifest = match parent_manifest {
+            ArcLocalChildManifest::File(_) => {
+                return Err(anyhow::anyhow!("parent id points to a file").into());
+            }
+            ArcLocalChildManifest::Folder(manifest) => manifest,
+        };
+
+        // Update the path result
+        let child_name = parent_manifest
+            .children
+            .iter()
+            .find_map(|(name, id)| {
+                if *id == current_entry_id {
+                    Some(name)
+                } else {
+                    None
+                }
+            })
+            .ok_or(ResolvePathError::EntryNotFound)?;
+        parts.push(child_name.clone());
+
+        // Update the confinement point result
+        // Give priority to the top most confinement point
+        confinement = if parent_manifest
+            .local_confinement_points
+            .contains(&current_entry_id)
+        {
+            PathConfinementPoint::Confined(current_parent_id)
+        } else {
+            confinement
+        };
```

Comment:
`Top most` is the furthest file from root (`/`)?

e.g.: `/foo/test.tmp` (furthest) vs `/foo.tmp` (nearest)

Reply:
```suggestion
        // Update the confinement point result
        // Give priority to the root most confinement point
        confinement = if parent_manifest
            .local_confinement_points
            .contains(&current_entry_id)
        {
            PathConfinementPoint::Confined(current_parent_id)
        } else {
            confinement
        };
```

Currently, it's prioritizing setting confinement to the entry nearest to the root (If that not wanted, adding an `or_else` should do the trick).

I propose replace `top` to `root` in the comment

Reply:
```mermaid
flowchart TD
    root --> folder1tmp--> folder2tmp  --> file
```

In that case the path confinement point would be folder1tmp ?

Reply:
> Currently, it's prioritizing setting confinement to the entry nearest to the root (If that not wanted, adding an or_else should do the trick).

Yes that's the behavior we want (first version of this PR got it wrong)

Reply:
> In that case the path confinement point would be folder1tmp ?

It would be root even, because it's the folder that tagged `folder1tmp` as confined

Reply:
> I propose replace top to root in the comment

Yes good point

Reply:
571c530cf9aacf14e2379652fc3b6187774559f0